service/src/main/java/bio/terra/workspace/service/job/JobBuilder.java

```diff
@@ -7,6 +7,7 @@
 import bio.terra.stairway.FlightMap;
 import bio.terra.workspace.common.utils.MdcHook;
 import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.job.JobService.JobResultOrException;
```

Comment:
there are some "async" calls for create and delete that we immediately return a jobid but the job may fail later. i don't want to do it in the flight because then i need to delete the entry in undostep(). and i think it's easier to forget. 

So I added a submitWithCallback so that we can control whether to log an update depending on the job result. @ddietterich

Reply:
There is nothing wrong with your submitWithCallback code, but I don't think it is the behavior we want.
If the pod fails, the operation happens, but your activity doesn't get logged.

My intended (but not implemented) solution to this was to add a special completion step to a flight. So after the flight is done - either failed or succeeded - Stairway would run the completion step. It would re-run the step until the completion step completed (including the usual retries, etc.)  That is also the key feature to enable notification instead of polling for async REST API calls.

So one option is, you could go do that! 😸

Another options would be something like writing the activity row with a start time but not an end time. Then when you get the callback from the flight, update the row with the end time (and maybe a success/fail indication).

Is there a reason you decided not to record failed activity?